antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/ANTLRFileStream.kt

```diff
@@ -5,23 +5,31 @@
  */
 package org.antlr.v4.kotlinruntime
 
-///**
-// * This is an [ANTLRInputStream] that is loaded from a file all at once
-// * when you construct the object.
-// *
-// */
-//@Deprecated("as of 4.7 Please use {@link CharStreams} interface.")
-//class ANTLRFileStream constructor(fileName: String, encoding: String? = null) : ANTLRInputStream() {
-//    override var sourceName: String
-//        protected set
-//
-//    init {
-//        this.sourceName = fileName
-//        load(fileName, encoding)
-//    }
-//
-//    fun load(fileName: String, encoding: String?) {
-//        data = Utils.readFile(fileName, encoding)
-//        this.n = data.size
-//    }
-//}
+import com.soywiz.korio.lang.Charset
```

Comment:
These imports surprise me

Reply:
There is a common lib by soywiz (a.k.a. korlibs) for IO utils, including multiplatform file reading, and the necessary struggles around it. Figured, might be good, since the code currently lacks a lot of IO features.

Reply:
Ok, good